lib/battle/CUnitState.h

```diff
@@ -141,6 +141,7 @@ class DLL_LINKAGE CUnitState : public Unit
 	bool drainedMana;
 	bool fear;
 	bool hadMorale;
+	SpellID usedSpell;
```

Comment:
Perhaps simple `castSpellThisTurn` will do? From what I see it is only used as a boolean.

Reply:
It's needed as SpellID in BattleFlowProcessor because functionality based on spell config.

Reply:
It's not. This was my first approach and not working. I've forgotten, why, but it's not the same (it's NONE at time of check).

Reply:
Don't see how this could be possible to be honest...

Reply:
Yes. You're right. My mistake.